hushline/settings.py

```diff
@@ -224,57 +224,55 @@ def index() -> str | Response:
         directory_visibility_form = DirectoryVisibilityForm()
         profile_form = ProfileForm()
 
-        # Check if the bio update form was submitted
-        if (
-            request.method == "POST"
-            and "update_bio" in request.form
-            and profile_form.validate_on_submit()
-        ):
-            user.bio = request.form["bio"]
-            user.extra_field_label1 = profile_form.extra_field_label1.data.strip()
-            user.extra_field_value1 = profile_form.extra_field_value1.data.strip()
-            user.extra_field_label2 = profile_form.extra_field_label2.data.strip()
-            user.extra_field_value2 = profile_form.extra_field_value2.data.strip()
-            user.extra_field_label3 = profile_form.extra_field_label3.data.strip()
-            user.extra_field_value3 = profile_form.extra_field_value3.data.strip()
-            user.extra_field_label4 = profile_form.extra_field_label4.data.strip()
-            user.extra_field_value4 = profile_form.extra_field_value4.data.strip()
-            db.session.commit()
-            flash("👍 Bio updated successfully.")
-            return redirect(url_for("settings.index"))
+        # Handle form submissions
+        if request.method == "POST":
+            # Update bio and custom fields
+            if "update_bio" in request.form and profile_form.validate_on_submit():
+                user.bio = request.form["bio"].strip()
+                user.extra_field_label1 = profile_form.extra_field_label1.data.strip()
+                user.extra_field_value1 = profile_form.extra_field_value1.data.strip()
+                user.extra_field_label2 = profile_form.extra_field_label2.data.strip()
+                user.extra_field_value2 = profile_form.extra_field_value2.data.strip()
+                user.extra_field_label3 = profile_form.extra_field_label3.data.strip()
+                user.extra_field_value3 = profile_form.extra_field_value3.data.strip()
+                user.extra_field_label4 = profile_form.extra_field_label4.data.strip()
+                user.extra_field_value4 = profile_form.extra_field_value4.data.strip()
+
+                # Trigger the rel=me verification for each URL field
```

Comment:
I think there's a clear case to be made that url verification would benefit from either parallelization (async coroutines) or deferred processing (celery workers). With 4 urls to verify, using coroutines to verify each url in parallel would roughly reduce the time by a factor of 4.

A deferred execution approach using something like celery would allow us to enqueue a verification task which would update the database with verification status, moving that work entirely outside the time the user is waiting for the next page to load.

There is a significant difference in the complexity of these two approaches. The parallelization approach would require adding an the `async` extra package to flask (https://flask.palletsprojects.com/en/3.0.x/async-await/) and some minor code changes. Adopting celery would add more complexity requiring an additional subsystem like Redis to act as a task broker and result store. 

I'd probably start with the async option here.

Reply:
@micahflee would value your take on this

Reply:
Yes I agree. At some point it seems inevitable that we'll need some sort of job queue. But for now, I think doing these async is a good approach.

Reply:
Changes in 90fadf7